### DISCUSSION TITLE
Problem rooms discover

### DISCUSSION THREAD

**Discussion by nakat0** [merdok/homebridge-miot#581] [General]
Good afternoon,

I am having problems with my xiaomi vacuum S12, when requesting the command:

miot send <IP> -t <TOKEN> get_room_mapping '[]'

![IMG_0050](https://github.com/merdok/homebridge-miot/assets/49118167/4464c680-57ee-4df8-915a-06f0af9baa58)

I can't get results, and I would like to be able to add rooms.

Thank you.

**Comment by merdok** [comment]
Maybe your device does not support that command.

**Reply by nakat0** [reply]
The robot works and is supported by the plugin, I can turn it on and off, activate vacuuming and/or mopping

The only thing I would need is to be able to add the rooms :(

**Reply by merdok** [reply]
As mentioned, it might be that your device does not support the `get_room_mapping` command. It could be a different command on your device...

**Reply by nakat0** [reply]
Is there any way to find out, what command can I use?

Thank you.

**Reply by merdok** [reply]
If there are no resources online for your device, then usually the best way is to sniff the traffic between the app and the device by using a modified mi home app.

**Reply by nakat0** [reply]
Ok, it's beyond my knowledge.

Let's wait a while and see, it is a model that is currently selling a lot and perhaps in a while its functions can be expanded.

Thank you.
